script.harmony.control/resources/lib/kodisettings.py

```diff
@@ -0,0 +1,54 @@
+#v.0.3.4
+
+from kodi_six import xbmc, xbmcaddon
+
+ADDON         = xbmcaddon.Addon()
+ADDONNAME     = ADDON.getAddonInfo('id')
+ADDONLONGNAME = ADDON.getAddonInfo('name')
+ADDONVERSION  = ADDON.getAddonInfo('version')
+ADDONPATH     = ADDON.getAddonInfo('path')
+ADDONDATAPATH = xbmc.translatePath( ADDON.getAddonInfo('profile') )
+ADDONICON     = xbmc.translatePath('%s/icon.png' % ADDONPATH )
+ADDONLANGUAGE = ADDON.getLocalizedString
+SKINNAME      = xbmc.getSkinDir()
+
+
+def _get_setting( setting_name, default, thetype="string" ):
+    if thetype.lower() == "bool":
+        if ADDON.getSetting( setting_name ).lower() == 'true':
+            return True
+        if ADDON.getSetting( setting_name ).lower() == 'false':
+            return False
+        return default
+    if thetype.lower() == "int":
+        try:
+            return int( ADDON.getSetting( setting_name ) )
+        except ValueError:
+            return default
+    if thetype.lower() == "number":
+        try:
+            return float( ADDON.getSetting( setting_name ) )
```

Comment:
You could use `ADDON.getSettingNumber()` it was introduced in Kodi Leia. See https://codedocs.xyz/AlwinEsch/kodi/group__python__xbmcaddon.html

Reply:
Honestly, I think I just left this as is to be consistent with all the other calls.